<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1200</width>
    <height>800</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1200</width>
    <height>800</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="mapLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>60</y>
      <width>401</width>
      <height>701</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="titleLabel">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>10</y>
      <width>1200</width>
      <height>32</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>1200</width>
      <height>32</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>1200</width>
      <height>32</height>
     </size>
    </property>
    <property name="text">
     <string>WELCOME TO 校园导航系统!!!</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="Log">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>180</y>
      <width>721</width>
      <height>441</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="countlabel">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>100</y>
      <width>241</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>地点数：0，边数：0</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignCenter</set>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>17</height>
    </rect>
   </property>
   <widget class="QMenu" name="add">
    <property name="title">
     <string>添加</string>
    </property>
    <addaction name="addPlace"/>
    <addaction name="addEdge"/>
   </widget>
   <widget class="QMenu" name="update">
    <property name="title">
     <string>更新</string>
    </property>
    <addaction name="separator"/>
    <addaction name="updatePlace"/>
    <addaction name="updateEdge"/>
   </widget>
   <widget class="QMenu" name="del">
    <property name="title">
     <string>删除</string>
    </property>
    <addaction name="delPlace"/>
    <addaction name="delEdge"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="openPlaceFile"/>
    <addaction name="openEdgeFile"/>
   </widget>
   <widget class="QMenu" name="menuchaxun">
    <property name="title">
     <string>查询</string>
    </property>
    <widget class="QMenu" name="menu_2">
     <property name="title">
      <string>地点</string>
     </property>
     <addaction name="searchForPlaceByName"/>
     <addaction name="searchForPlacesByType"/>
     <addaction name="searchByTime"/>
    </widget>
    <addaction name="menu_2"/>
    <addaction name="searchForEdge"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>操作</string>
    </property>
    <addaction name="showAllPlaces"/>
    <addaction name="showAllEdges"/>
    <addaction name="resetGraph"/>
    <addaction name="printGraph"/>
    <addaction name="clearLog"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>算法</string>
    </property>
    <addaction name="isConnected"/>
    <addaction name="hasEulerPath"/>
    <addaction name="showEulerPath"/>
    <addaction name="shortestPath"/>
    <addaction name="findMST"/>
   </widget>
   <addaction name="add"/>
   <addaction name="update"/>
   <addaction name="del"/>
   <addaction name="menu"/>
   <addaction name="menuchaxun"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="addPlace">
   <property name="text">
    <string>地点</string>
   </property>
  </action>
  <action name="addEdge">
   <property name="text">
    <string>路径</string>
   </property>
  </action>
  <action name="updatePlace">
   <property name="text">
    <string>地点</string>
   </property>
  </action>
  <action name="updateEdge">
   <property name="text">
    <string>路径</string>
   </property>
  </action>
  <action name="delPlace">
   <property name="text">
    <string>地点</string>
   </property>
  </action>
  <action name="delEdge">
   <property name="text">
    <string>路径</string>
   </property>
  </action>
  <action name="searchForEdge">
   <property name="text">
    <string>路径</string>
   </property>
  </action>
  <action name="searchForPlaceByName">
   <property name="text">
    <string>按名称</string>
   </property>
  </action>
  <action name="searchForPlacesByType">
   <property name="text">
    <string>按类别</string>
   </property>
  </action>
  <action name="openPlaceFile">
   <property name="text">
    <string>读取地点文件</string>
   </property>
  </action>
  <action name="openEdgeFile">
   <property name="text">
    <string>读取道路文件</string>
   </property>
  </action>
  <action name="showAllPlaces">
   <property name="text">
    <string>显示所有地点信息</string>
   </property>
  </action>
  <action name="showAllEdges">
   <property name="text">
    <string>显示所有道路信息</string>
   </property>
  </action>
  <action name="resetGraph">
   <property name="text">
    <string>重置图信息</string>
   </property>
  </action>
  <action name="printGraph">
   <property name="text">
    <string>打印图信息</string>
   </property>
  </action>
  <action name="clearLog">
   <property name="text">
    <string>清空日志信息</string>
   </property>
  </action>
  <action name="isConnected">
   <property name="text">
    <string>判断图的连通性</string>
   </property>
  </action>
  <action name="hasEulerPath">
   <property name="text">
    <string>判断是否存在欧拉路径</string>
   </property>
  </action>
  <action name="showEulerPath">
   <property name="text">
    <string>打印欧拉路径</string>
   </property>
  </action>
  <action name="shortestPath">
   <property name="text">
    <string>计算两点最短路径</string>
   </property>
  </action>
  <action name="searchByTime">
   <property name="text">
    <string>按建议时间</string>
   </property>
  </action>
  <action name="findMST">
   <property name="text">
    <string>求最小生成树</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
